src/components/layer-timestamp-card/time-slider.vue

```diff
@@ -0,0 +1,87 @@
+<template>
+  <div class="time-slider">
+    <v-row>
+      <v-col cols="2" class="my-auto">
+        <slot v-if="timeIndex !== 0" :back="back" name="backButton" />
+      </v-col>
+      <v-col cols="8">
+        <slot name="label" />
+      </v-col>
+      <v-col cols="2" class="my-auto">
+        <slot
+          v-if="timeIndex !== dates.length - 1"
+          :forward="forward"
+          name="forwardButton"
+        />
+      </v-col>
+    </v-row>
+  </div>
+</template>
+
+<script>
+/* import _ from "lodash"; */
+
+export default {
+  props: {
+    dates: {
+      type: Array,
+      default: () => [],
+    },
+    startAt: {
+      // Where to start in the dates array, begin end or at random index
+      default: () => "start",
+      validator(value) {
+        if (Number.isInteger(value)) {
+          return true;
+        } else if (["start", "end"].includes(value)) {
+          return true;
+        } else {
+          return false;
+        }
+      },
+    },
+    setTimeIndex: {
+      type: Number,
+      default: () => 0,
+    },
+  },
+  data() {
+    return {
+      timeIndex: 0,
+    };
+  },
+  computed: {
+    currentDateObject() {
+      return this.dates[this.timeIndex];
+    },
+  },
+  mounted() {
+    this.setInitialTimeIndex();
+  },
+  methods: {
+    setInitialTimeIndex() {
+      // Set the timeIndex when timeslider is mounted or dates have changed
+      if (!this.dates) {
+        return;
+      }
+      if (this.startAt === "end" && this.dates.length > 0) {
+        this.timeIndex = this.dates.length - 1;
+      } else if (Number.isInteger(this.startAt)) {
+        this.timeIndex = this.startAt;
+      } else {
+        this.timeIndex = 0;
+      }
+    },
+    forward() {
```

Comment:
forward and back could be one function (step could then either be +1 or -1)

update (step) {
    this.timeIndex = this.timeIndex + step 


But I can imagine that you find this easier to read, so no need of implementing, just a suggestion